### ISSUE TITLE
[Suggestion] Autostart & autoclose command line switches

### ISSUE THREAD

**Issue by cragrim** [superzanti/ServerSync#81]
#### Serversync Version: 
serversync-2.6.9.jar

#### Minecraft Version: 
1.11.2

#### Issue: 

First I'd like to say THANK YOU its working BEAUTIFULLY <3

This is not an actual issue, just a minor convenience suggestion/question. This might already be implemented but I could not find it when searching for it. 

If its not too much work I would love some optional command line arguments for starting syncing automatically and another optional command line for closing the window automatically once completed.

I'm currently running ServerSync at PreLaunch in a MultiMC instance:
_PreLaunchCommand=java -jar "$INST_MC_DIR/serversync-2.6.9.jar"_

So with some additional command line switches:
_PreLaunchCommand=java -jar **-autostart -autoclose** "$INST_MC_DIR/serversync-2.6.9.jar"_

Thanks again !
Best Regards

**Comment by rheimus** [comment]
From 2.6.13.1 onwards there are two flags available.
- progress-only : auto start/finish of SS with the GUI available to track progress
- silent : auto start/finish of SS as a fully headless background process

`
java -jar "$INST_MC_DIR/serversync-2.6.9.jar silent"
java -jar "$INST_MC_DIR/serversync-2.6.9.jar progress-only"
`

**Comment by rheimus** [comment]
Have updated the [WIKI](https://github.com/superzanti/ServerSync/wiki) with a more detailed description of the command line options, under the automation section

**Comment by cragrim** [comment]
Nice work!

I seem to be having trouble with the progress-only switch though.
No window shows up when I use it and nothing is updated.

**I run this batch example to compare with and without using the switch:**
_java -jar serversync-2.6.9.jar
pause
java -jar serversync-2.6.9.jar progress-only
pause_

Gif video:
![Gif](http://i.imgur.com/80vkrK3.gif)

See how nothing happens the second time?
Maybe the switch wasnt implemented for MC V1.11.2 i.e. serversync-2.6.9 ?

Also I'm having trouble with MultiMC using switches. It appears to use some kind of string formatting on the launch command. Can I set the progress-only arg from a setting in the config file instead?

Really appreciate the help ! 
Cheers :)

**Comment by cragrim** [comment]
I read your post again and realized it was not implemented in the version I was using so I tried to connect using serversync-2.6.15 on the client to the server running serversync-2.6.9 in the mod folder.

This unfortunately resulted in the server crashing. I guess I have to use the method of having the server ServerSync running as a separate process? (I.e. not in the mod folder)

**Comment by rheimus** [comment]
SS should work on any recent version of MC, though I cant recall if I tested it on 11.2.

But yes it will work using the non forge loaded method.

**Comment by rheimus** [comment]
I'll have a look into MultiMc and 11.2 see what I can dig up

**Comment by cragrim** [comment]
Here you go! Had to get back home and restart stuff.

Basically what happens is when you try to connect the server leaks memory and uses more and more ram until all ram is filled and it crashes. It kept dumping errors into the log. Hence the log file grew to 50 MB in a couple of seconds. I zipped all logs before and after connecting below:

**ServerSync 2.6.15 Log:**
_< Connecting to server >
Checking config...
The mods between server and client are incompatable... Updating...
<------> Getting files <------>
Getting mods...
Failed to get files from server, check detailed log in minecraft/logs
Errors occurred, please check ip/port details are correct. For a detailed log check the logs folder in your minecraft directory_


**Logs before the crash:**
https://drive.google.com/file/d/0BxIgRHQZ_J9Hdjl3WXhybDQ0M28/view?usp=sharing

**Logs after the crash**
https://drive.google.com/file/d/0BxIgRHQZ_J9HQ3ktanU2eGpFOUU/view?usp=sharing

**Comment by cragrim** [comment]
I think its a good idea to always run the server separately never the less and avoid having it in the mod folder. I just need to figure out how to do that.

**Comment by cragrim** [comment]
I set up standalone serversync-2.6.15.jar with the batch command you supplied in the wiki!
```
@echo off
SETLOCAL
for %%f in (forge*.jar) do (SET forge=%%~nxf)
for %%f in (serversync*.jar) do (SET serversync=%%~nxf)
start "ServerSync Server" java -jar %serversync% server
ENDLOCAL
```

**With the following PreLaunch command in my MultiMC instance:**
_java -jar "$INST_MC_DIR/serversync-2.6.15.jar" "progress-only"_

Its now working beautifully automatically updating without needing to press sync or closing the window!
Thanks again, this is awesome and saves so much headache keeping modpack up to date!

**Comment by rheimus** [comment]
@cragrim Interesting, for my MultiMc I had to tell it to reach into the minecraft folder of the instance.

````batchfile
java -jar "$INST_DIR/minecraft/serversync-2.6.15.jar" progress-only
````

**Comment by rheimus** [comment]
It's possible this is forge fiddling with classes, I can not reproduce memory leaks on a test 11.2 server with SS 2.6.15.

Only thing worthy of note was that forge attempted to rewrite parts of SS that use System.exit(), which do not get run in server mode anyway.

Also could not access the log files, get some 0 byte zips.

**Comment by rheimus** [comment]
Though looking a the server worker it does look like there is a potential for some resource leaks. Nothing that would cause OOM unless masses of connections were attempted though.

**Comment by cragrim** [comment]
Thanks for being patient with my hopeless ramblings. Somehow google drive broke the zip files with the logs, I tested downloading them and totally blame google for that. Running ServerSync server separate outside of forge is the obvious optimal solution! There is no point in using the old version from inside the mod folder. Having several people syncing/downloading at the same time would probably have caused unnecessary server lag anyway.

Everything is working beautifully now as an autostarting sync in MultiMC. I can even have my mother use and update the modpack! Now thats something

Thanks for all your help, youve made updating private modpacks so much easier!
Closing this issue.
